File: Gomoku-Judge-update/bot.java
import java.util.*;
import org.json.simple.JSONValue;

class Main {
	static class Grid {
		public static final int SIZE = 15;
		private int[][] grid = new int[SIZE][SIZE];
		public void placeAt(int x, int y) {
			if (x >= 0 && y >= 0) {
				grid[x][y] = 1;
			}
		}
		public Map randomAvailablePosition() {
			List xList = new LinkedList();
			List yList = new LinkedList();
			for (int i = 0; i < SIZE; i++) {
				for (int j = 0; j < SIZE; j++) {
					if (grid[i][j] == 0) {
						xList.add(i);
						yList.add(j);
					}
				}
			}
			int rand = (int) (Math.random()*xList.size());
			Map map = new HashMap();
			map.put("x", xList.get(rand));
			map.put("y", yList.get(rand));
			return map;
		}
	}
	public static void main(String[] args) {
		String input = new Scanner(System.in).nextLine();
		Map<String, List> obj = (Map) JSONValue.parse(input);
		Grid grid = new Grid();
		List<Map> requests = obj.get("requests");
		List<Map> responses = obj.get("responses");
		for (Map<String, Long> rec : requests) {
			grid.placeAt(rec.get("x").intValue(), rec.get("y").intValue());
		}
		for (Map<String, Long> rec : responses) {
			grid.placeAt(rec.get("x").intValue(), rec.get("y").intValue());
		}
		Map output = new HashMap();
		output.put("response", grid.randomAvailablePosition());
		System.out.print(JSONValue.toJSONString(output));
	}
}

File: Gomoku-Judge-update/bot.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cstdlib>
#include <ctime>
#include "jsoncpp/json.h" // C++编译时默认包含此库

using namespace std;

const int SIZE = 15;
bool Grid[SIZE][SIZE];

void placeAt(int x, int y)
{
	if (x >= 0 && y >= 0) {
		Grid[x][y] = true;
	}
}
Json::Value randomAvailablePosition()
{
	vector<int> xs, ys;
	for (int i = 0; i < SIZE; ++i) {
		for (int j = 0; j < SIZE; ++j) {
			if (Grid[i][j] == false) {
				xs.push_back(i);
				ys.push_back(j);
			}
		}
	}
	srand(time(0));
	int rand_pos = rand() % xs.size();
	Json::Value action;
	action["x"] = xs[rand_pos];
	action["y"] = ys[rand_pos];
	return action;
}
int main()
{
	// 读入JSON
	string str;
	//fstream cin("in.txt", ios_base::in | ios_base::out);
	getline(cin, str);
	Json::Reader reader;
	Json::Value input;
	reader.parse(str, input);

	// 分析自己收到的输入和自己过往的输出，并恢复状态
	int turnID = input["responses"].size();
	for (int i = 0; i < turnID; i++) {
		placeAt(input["requests"][i]["x"].asInt(), input["requests"][i]["y"].asInt());
		placeAt(input["responses"][i]["x"].asInt(), input["responses"][i]["y"].asInt());
	}
	placeAt(input["requests"][turnID]["x"].asInt(), input["requests"][turnID]["y"].asInt());
	// 做出决策存为myAction
	// 输出决策JSON
	Json::Value ret;
	ret["response"] = randomAvailablePosition();
	Json::FastWriter writer;
	cout << writer.write(ret) << endl;
	return 0;
}
